Developers/micropayments.md

```diff
@@ -0,0 +1,130 @@
+# Trustless micropayment channels
+
+## Protocol
+
+![Micropayment Protocol](micropayment_protocol.png)
+
+
+### Deposit Transaction
+
+The deposit transaction ensures the payee cannot spend the funds while
+the channel is open.
+
+The pay to script allows for three possible outcoms:
+
+ - The payer can recover the deposit after a timeout if no commit was made.
+ - Both parties commit and close the channel.
+ - The payer collects the change after the payee publishes the spend secret.
+
+**The commit transaction spend secret must be the same as the deposit
+transaction spend secret!**
+
+pay to script:
+
+    OP_IF
+        2 <payer pubkey> <payee pubkey> 2 OP_CHECKMULTISIG
+    OP_ELSE
+        OP_IF
+            OP_HASH160 <spend secret hash> OP_EQUALVERIFY
+            <payer pubkey> OP_CHECKSIG
+        OP_ELSE
+            <expire time> OP_CHECKSEQUENCEVERIFY OP_DROP
+            <payer pubkey> OP_CHECKSIG
+        OP_ENDIF
+    OP_ENDIF
+
+
+### Expire Transaction
+
+The expire transaction is used by the payee to recover the and funds after
```

Comment:
Would it make sense to use source and destination instead of payer and payee?
The rest of docs uses the former terms, but I don't know if those are suitable for micropayments.